File: sample/checker.c
/*
 * @(#)checker.c
 * $Id$
 */

#include <stdio.h>
#include <stdlib.h>
#include "license.h"

int main(int argc, char* argv[])
{
  if (argc < 2) {
    printf("usage: checker [license] [version]\n");
    exit(-1);
  }

  if (license_is_valid_file(argv[1], argv[2])) {
    printf("valid!\n");
    return 0;
  } else {
    printf("invalid!\n");
    return -1;
  }
}

/* the end of file */

File: sample/Checker.java
/*
 * @(#)Checker.java
 * $Id$
 */

import jp.co.ipride.License;


public class Checker {
    public static void main(String[] args) {
	if (args.length < 2) {
	    System.out.println("usage: Checker [license] [version]\n");
	    System.exit(-1);
	}

	if (License.isValidFile(args[0], args[1])) {
	    System.out.println("valid!");
	    System.exit(0);
	} else {
	    System.out.println("invalid!");
	    System.exit(-1);
	}
    }
}

/* the end of file */
